frontend/src/scenes/userLogic.ts

```diff
@@ -76,6 +77,20 @@ export const userLogic = kea<userLogicType>([
                         return values.user
                     }
                 },
+                deleteUser: async () => {
+                    return await api
+                        .delete('api/users/@me/')
```

Comment:
This is potentially overkill but I have seen patterns before where its like "/request_deletion_code" which returns a value (like a time bound hashed sha of the user name or something) and then the actual request requires that code which can be good for essentially building double confirmation into the api itself.

This PR is so good that I don't think its necessary but food for thought in the future.